src/python/pants/option/arg_splitter.py

```diff
@@ -168,37 +152,35 @@ def assign_flag_to_scope(flg: str, default_scope: str) -> None:
             assign_flag_to_scope(flag, GLOBAL_SCOPE)
         scope, flags = self._consume_scope()
         while scope:
-            if not self._check_for_help_request(scope.lower()):
-                add_scope(scope)
-                if scope in self._known_goal_scopes:
-                    goals.add(scope)
-                else:
-                    unknown_scopes.append(scope)
-                for flag in flags:
-                    assign_flag_to_scope(flag, scope)
+            add_goal(scope)
+            for flag in flags:
+                assign_flag_to_scope(flag, scope)
             scope, flags = self._consume_scope()
 
         while self._unconsumed_args and not self._at_double_dash():
-            arg = self._unconsumed_args.pop()
-            if arg.startswith("-"):
+            if self._at_flag():
+                arg = self._unconsumed_args.pop()
                 # We assume any args here are in global scope.
-                if not self._check_for_help_request(arg):
-                    assign_flag_to_scope(arg, GLOBAL_SCOPE)
-            elif self.likely_a_spec(arg):
+                assign_flag_to_scope(arg, GLOBAL_SCOPE)
+                continue
+
+            arg = self._unconsumed_args.pop()
+            if self.likely_a_spec(arg):
                 specs.append(arg)
-            elif arg not in self._known_scopes:
-                unknown_scopes.append(arg)
+            else:
+                add_goal(arg)
 
         if self._at_double_dash():
             self._unconsumed_args.pop()
             passthru = list(reversed(self._unconsumed_args))
 
-        if unknown_scopes and not self._help_request:
-            self._help_request = UnknownGoalHelp(tuple(unknown_scopes))
+        if unknown_scopes and "__unknown_goal" in self._known_goal_scopes:
+            goals.add("__unknown_goal")
 
-        if not goals and not self._help_request:
-            self._help_request = NoGoalHelp()
+        if not goals and "__no_goal" in self._known_goal_scopes:
+            goals.add("__no_goal")
```

Comment:
Contemplating whether these checks belong here at all.. also I'm not very fond of the hardcoded goal names for it.. but haven't come up with a better approach for it, yet...